### PR TITLE
[WIP][Do Not Submit] Experimental gRPC-Swift support

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: FirebaseFirestore.podspec [hunk 1]
```diff
@@ -19,6 +19,7 @@ Google Cloud Firestore is a NoSQL document database built for automatic scaling,
   s.ios.deployment_target = '10.0'
   s.osx.deployment_target = '10.12'
   s.tvos.deployment_target = '10.0'
+  s.watchos.deployment_target = '3.0'
```
Comment (paulb777) [inline FirebaseFirestore.podspec:22]: 6.0 to be consistent with Firebase

File: FirebaseFirestore.podspec [hunk 2]
```diff
@@ -98,7 +99,7 @@ Google Cloud Firestore is a NoSQL document database built for automatic scaling,
   s.dependency 'abseil/time', abseil_version
   s.dependency 'abseil/types', abseil_version
 
-  s.dependency 'gRPC-C++', '~> 1.28.0'
+  s.dependency 'GRPCSwiftShim', '~> 0.0.1'
```
Comment (paulb777) [inline FirebaseFirestore.podspec:102]: Why is the Shim a separate library versus part of Firestore itself?

File: GRPCSwiftShim.podspec [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#
+# Be sure to run `pod lib lint FirebaseFirestoreSwift.podspec' to ensure this is a
```
Comment (paulb777) [inline GRPCSwiftShim.podspec:2]: Add the `pod lib lint` to CI

File: GRPCSwiftShim.podspec [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  s.license                 = { :type => 'Apache', :file => 'LICENSE' }
+  s.authors                 = 'Google, Inc.'
+
+  s.source                  = {
+    :git => 'https://github.com/Firebase/firebase-ios-sdk.git',
+    :tag => 'CocoaPods-' + s.version.to_s
+  }
+
+  s.swift_version           = '5.3'
+  s.ios.deployment_target   = '11.0'
```
Comment (paulb777) [inline GRPCSwiftShim.podspec:25]: Should be 10.0

File: GRPCSwiftShim.podspec [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  s.osx.deployment_target   = '10.12'
+  s.tvos.deployment_target  = '11.0'
```
Comment (paulb777) [inline GRPCSwiftShim.podspec:27]: add watchos

**Review by paulb777 — COMMENTED** [review]
looks promising! Just a few comments/questions on library structure for now.

File: FirebaseFirestore.podspec (hunk 1 above)
Comment (wu-hui) [reply FirebaseFirestore.podspec]: Done.

File: GRPCSwiftShim.podspec (hunk 5 above)
Comment (wu-hui) [reply GRPCSwiftShim.podspec]: Done.

File: GRPCSwiftShim.podspec (hunk 4 above)
Comment (wu-hui) [reply GRPCSwiftShim.podspec]: Done.

File: FirebaseFirestore.podspec (hunk 2 above)
Comment (wu-hui) [reply FirebaseFirestore.podspec:103]: Because it is just a quick hack prototype.

**Comment by wu-hui** [comment]
> looks promising! Just a few comments/questions on library structure for now.

Thanks for the review! This is far from ready though, I am mostly hacking something together quick and dirty.

**Pull request closed without merging**